src/Components/animation/ProgressBar.js

```diff
@@ -11,226 +16,235 @@ class ProgressBar extends React.Component {
   // - animationWidth/animationHeight: size in pixel of the animation
   // - stepCount: the number of steps
   // - currentStepAV: an animated value which represents the current step
-  // - getStepAnimation(): set the animation to a given step
   // - stepLength: length of an animation step in ms
   constructor(props) {
     super(props);
 
     this.state = {
-      animationWidth: 1,
-      animationHeight: 1,
+      stateFromProps: _initializeStateFromProps(props),
     };
+  }
 
-    // Opacity of the step dots (visible if the current step is >= to the step represented by the dot)
-    this.dotsOpacity = [];
+  _addStepButtonClicked() {
+    if (this.props.onStepAdded() !== undefined && this.props.onStepAdded !== null) this.props.onStepAdded();
+  }
 
-    for (var i = 0; i < this.props.stepCount; i++)
-      // The first step is always visible
-      if (i === 0) this.dotsOpacity.push(new Animated.Value(1));
-      else this.dotsOpacity.push(new Animated.Value(0));
+  render() {
+    return (
+      <View style={[{ position: 'absolute', left: 0, top: 0 }]} height="100%" width="100%">
+        {/* Gray dots */}
+        {this.props.readonly
```

Comment:
You can rewrite the condition `{!this.props.readlonly && this.state.state...}` it is less verbose

Reply:
Ok (this syntax feels strange but I am all for consision)